src/sklearndf/wrapper/_wrapper.py

```diff
@@ -824,6 +798,83 @@ def _prediction_with_class_labels(
             raise TypeError(f"unexpected type or prediction result: {type(y).__name__}")
 
 
+# noinspection PyPep8Naming
+@inheritdoc(match="[see superclass]")
+class ClustererWrapperDF(
+    ClustererDF,
+    LearnerWrapperDF[T_NativeClusterer],
+    Generic[T_NativeClusterer],
+    metaclass=ABCMeta,
+):
+    """
+    Base class of DF wrappers for native clusterers conforming with the scikit-learn
+    API.
+
+    Clusterer wrapper classes should be created using function
+    :func:`.make_df_clusterer`.
+    """
+
+    COL_LABELS = "labels"
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self._x_index: Optional[pd.Index] = None
+
+    @property
+    def labels_(self) -> pd.Series:
+        """[see superclass]"""
+        self._ensure_fitted()
+        raw_labels = self._native_estimator.labels_
+
+        return pd.Series(data=raw_labels, name=self.COL_LABELS, index=self._x_index)
+
+    def _post_fit(
+        self,
+        X: pd.DataFrame,
+        y: Optional[Union[pd.Series, pd.DataFrame]] = None,
+        **fit_params,
+    ) -> None:
+        super()._post_fit(X, y, **fit_params)
+        self._x_index = X.index
+
+    def _reset_fit(self) -> None:
+        super()._reset_fit()
+        self._x_index = None
+
+    def fit_predict(
```

Comment:
Just for cosmetics, I would define public methods first and move the protected ones further down.

Reply:
Sure, reordered.